lib/database/repositories/EnvironmentHistoryItemRepository.js

```diff
@@ -11,8 +11,35 @@
  * or submit itself to any jurisdiction.
  */
 
-const { models: { EnvironmentHistoryItem } } = require('../');
+const { models: { EnvironmentHistoryItem }, sequelize } = require('../');
 const Repository = require('./Repository');
+const { timestampToMysql } = require('../../server/utilities/timestampToMysql.js');
+
+/**
+ * Returns the SQL query for fetching the environment history combinations and
+ * their occurrences within a specified time period.
+ *
+ * @param {Period} period the period on which the occurrences must be counted
+ * @return {string} the raw SQL query
+ */
+const getHistoryDistributionQuery = ({ from, to }) => `
+    SELECT environments, COUNT(*) AS count
+    FROM (
+        SELECT e.id, GROUP_CONCAT(ehi.status) AS environments
+        FROM environments e
+        JOIN environments_history_items ehi ON e.id = ehi.environment_id
+        JOIN (
+            SELECT DISTINCT environment_id
+            FROM environments_history_items
+            WHERE created_at BETWEEN '${timestampToMysql(from)}' AND '${timestampToMysql(to)}'
+        ) AS environment_ids_in_period 
```

Comment:
Actually, I think you should symply check that the environment has `created_at` before the `to` and `updated_at` after the `from`

Reply:
You're entirely right. Now that I think about this approach, it should be more efficient and less complicated.